internal/postprocessor/main.go

```diff
@@ -268,3 +246,177 @@ func docURL(cloudDir, importPath string) (string, error) {
 	pkgPath := strings.TrimPrefix(strings.TrimPrefix(importPath, mod), "/")
 	return "https://cloud.google.com/go/docs/reference/" + mod + "/latest/" + pkgPath, nil
 }
+
+func (c *config) AmendPRDescription(ctx context.Context) error {
+	log.Println("Amending PR title and body")
+	pr, err := c.getPR(ctx)
+	if err != nil {
+		return err
+	}
+	prTitle := pr.Title
+	prBody := pr.Body
+	newPRTitle, newPRBody, err := c.processCommit(*prTitle, *prBody)
+	if err != nil {
+		return err
+	}
+	if err := c.writePRCommitToFile(newPRTitle, newPRBody); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *config) processCommit(title, body string) (string, string, error) {
+	var newPRTitle string
+	var commitTitle string
+	var commitTitleIndex int
+
+	bodySlice := strings.Split(body, "\n")
+	for index, line := range bodySlice {
+		if strings.Contains(line, "[REPLACEME]") {
+			commitTitle = line
+			commitTitleIndex = index
+			continue
+		}
+		if !strings.Contains(line, "googleapis/googleapis/") {
+			continue
+		}
+		commitPkg, err := c.getScope(line)
+		if err != nil {
+			return "", "", err
+		}
+
+		if newPRTitle == "" {
+			newPRTitle = updateCommitTitle(title, commitPkg)
+			continue
+		}
+		newCommitTitle := updateCommitTitle(commitTitle, commitPkg)
+		bodySlice[commitTitleIndex] = newCommitTitle
+	}
+	body = strings.Join(bodySlice, "\n")
+	return newPRTitle, body, nil
+}
+
+func (c *config) getPR(ctx context.Context) (*github.PullRequest, error) {
+	client := github.NewClient(nil)
+	prs, _, err := client.PullRequests.List(ctx, c.githubUsername, "google-cloud-go", nil)
+	if err != nil {
+		return nil, err
+	}
+	pr, err := c.findValidPR(ctx, prs)
+	if err != nil {
+		return nil, err
+	}
+	return pr, nil
+}
+
+func (c *config) findValidPR(ctx context.Context, prs []*github.PullRequest) (*github.PullRequest, error) {
+	for _, pr := range prs {
+		if strings.Contains(*pr.Head.Label, c.branchPrefix) {
+			owlbotPR := pr
+			return owlbotPR, nil
+		}
+	}
+	return nil, errors.New("no PR found")
+}
+
+func (c *config) getScope(line string) (string, error) {
+	hash := extractHashFromLine(line)
+	commitPkg, err := c.getScopeFromGoogleapisCommitHash(hash)
```

Comment:
nit: function is called getScope but return value is called commitPkg. Maybe call it scope? Also I migth suggest inlining this function

Reply:
Done.